File: angular-employee-management/src/app/models/register-model.ts
import { RegisterPhoneModel } from "./register-phone-model";
import { Role } from "./role.enum";

export class RegisterModel {
    email: string | undefined;
    firstName: string | undefined;
    lastName: string | undefined;
    docNumber: string | undefined;
    docType: number | undefined;
    phones: RegisterPhoneModel[] = [];
    birthDate: string | undefined;
    managerId?: string;
    password: string | undefined;
    role: Role | undefined;
}

File: src/Application.Models/Employees/RegisterModel.cs

using System.ComponentModel.DataAnnotations;

namespace Application.Models.Employees;

public class RegisterModel
{
    public string Id { get; set; } = Guid.NewGuid().ToString();

    [Required]
    public string Email { get; set; }

    [Required]
    public string FirstName { get; set; }
    
    [Required]
    public string LastName { get; set; }
    
    [Required]
    public string DocNumber { get; set; }
    
    [Required]
    public short DocType { get; set; }
    
    [Required]
    public List<RegisterPhoneModel> Phones { get; set; } = new List<RegisterPhoneModel>();
    
    [Required]
    public DateTime BirthDate { get; set; }
    
    [Required]
    public string ManagerId { get; set; }
        
    [Required]
    public string Password { get; set; }
    
    [Required]
    public short Role { get; set; }
}
